[Bug]: squads run fails with Google/Gemini Provider due to Worktree and Headless CLI security policies

### Bug Description

When attempting to run an agent using the Google provider (via @google/gemini-cli), the execution fails immediately. The issue stems from a conflict between squads-cli's sandboxing architecture (Git Worktrees /
  background execution) and the strict security policies built into the gemini CLI.

  Specifically, two separate security mechanisms in Gemini CLI block the execution:
   1. File System Sandboxing (`Path not in workspace`): squads-cli creates a temporary isolated git worktree for the agent but instructs the agent to read its markdown definition file located in the original
      project root. Gemini CLI blocks this cross-directory read for security reasons.
   2. Interactive Confirmation Enforcement (`Tool execution denied by policy`): squads-cli runs the gemini command in the background (headless mode) to execute tools. Because Gemini CLI is designed to prompt the user for confirmation (Y/n) before running shell commands, and it cannot do so in the background, it defaults to denying the execution entirely.


### Steps to Reproduce

   1. Initialize a squads environment (squads init).
   2. Set provider to google in .agents/config/provider.yaml.
   3. Create a basic squad and agent (e.g., product/product-manager.md).
   4. Run the squad: squads run product "test"

### Expected Behavior

When running squads run <squad> [agent] with provider: google configured, the gemini CLI should:


   1. Successfully read the agent's definition file (e.g., .agents/squads/product/product-manager.md) without triggering a "Path not in workspace" security error, even when executed from within the isolated
      .worktrees directory created by squads-cli.
   2. Execute the assigned tasks (shell commands, file edits) autonomously in the background, without getting blocked by a "Tool execution denied by policy" error. This implies that squads-cli should pass the
      appropriate auto-approval flags (like --yolo or --approval-mode yolo) to the gemini CLI when spawning it in headless mode.
   3. Complete the agent's pipeline (e.g., creating the PRODUCT_SPEC.md or opening a PR) and exit cleanly, returning the results to the squads orchestrator.

### Actual Behavior

When running squads run <squad> [agent] with provider: google, the execution fails immediately. The output in the terminal shows a cascade of Gemini CLI security errors:


   1. Workspace/Worktree Path Conflict:

Running product-manager with Google...Error executing tool read_file: Path not in workspace: Attempted path "/Users/myuser/project/.agents/squads/product/product-manager.md" resolves outside the allowed

     workspace directories: /Users/myuser/.worktrees/product-product-manager-1771882752908 or the project temp directory: /Users/myuser/.gemini/tmp/product-product-manager-1771882752908
     Explanation: squads-cli spawns the gemini process inside a newly created isolated git worktree (/Users/.../.worktrees/...), but instructs it to read its agent definition file located in the original project
  root. Gemini CLI blocks this cross-directory read for security reasons.

   2. Headless Tool Execution Denied:


Running product-manager with Google...Error executing tool shell_execute: Tool execution denied by policy.
Running product-manager with Google...Error executing tool terminal_operate: Tool execution denied by policy.
     Explanation: By design, Gemini CLI operates interactively and requires explicit user confirmation (Y/n) in the terminal before executing any shell commands. Since squads-cli spawns the gemini command in the
  background/headless mode, Gemini cannot prompt the user for confirmation and automatically denies the tool execution as a fail-safe.

### Squads CLI Version

0.6.2

### Node.js Version

v25.6.1

### Operating System

macOS

### Relevant Logs

```shell
$ squads run Product

  Started: 2026-02-23T21:39:12.860Z

  Pipeline product-manager → system-architect → issue-writer

[1/3]
⠧ Running product-manager with Google...Both GOOGLE_API_KEY and GEMINI_API_KEY are set. Using GOOGLE_API_KEY.
[ERROR] [IDEClient] Failed to connect to IDE companion extension. Please ensure the extension is running. To install the extension, run /ide install.
Loading extension: firebase
Loading extension: firebase-rules
Loading extension: github
Skipping MCP config for server with name "github" from extension "github" as it already exists.
MCP server 'github' requires authentication using: /mcp auth githubServer 'primeng' supports tool updates. Listening for changes...
Server 'nx-mcp' supports tool updates. Listening for changes...
Server 'nx-mcp' supports resource updates. Listening for changes...
Server 'firebase' supports tool updates. Listening for changes...
Server 'firebase' supports prompt updates. Listening for changes...
[... theme registration logs omitted ...]
Both GOOGLE_API_KEY and GEMINI_API_KEY are set. Using GOOGLE_API_KEY.
⠧ Running product-manager with Google...Error executing tool read_file: File not found.
⠸ Running product-manager with Google...Error executing tool read_file: Path not in workspace: Attempted path "/[REDACTED_PATH]/.agents/squads/product/product-manager.md" resolves outside the allowed workspac
directories: /[REDACTED_PATH]/../.worktrees/product-product-manager-[TIMESTAMP] or the project temp directory: /[REDACTED_PATH]/../.gemini/tmp/product-product-manager-[TIMESTAMP]
⠏ Running product-manager with Google...Error executing tool shell_execute: Tool execution denied by policy.
⠴ Running product-manager with Google...Error executing tool shell_execute: Tool execution denied by policy.
```

### Additional Context

_No response_

### Checklist

- [x] I have searched existing issues to avoid duplicates
- [x] I have provided all required information above
- [x] I am willing to help test a fix if needed

--- a/src/commands/run.ts
+++ b/src/commands/run.ts
@@ -1,7 +1,7 @@
 import ora from 'ora';
 import { spawn, execSync } from 'child_process';
 import { join, dirname } from 'path';
-import { existsSync, readFileSync, writeFileSync, mkdirSync, readdirSync } from 'fs';
+import { existsSync, readFileSync, writeFileSync, mkdirSync, readdirSync, cpSync } from 'fs';
 import {
   findSquadsDir,
   loadSquad,
@@ -2224,7 +2224,6 @@ async function executeWithProvider(
   }
 
   const projectRoot = options.cwd || getProjectRoot();
-  const args = cliConfig.buildArgs(prompt);
   const squadName = options.squadName || 'unknown';
   const agentName = options.agentName || 'unknown';
   const timestamp = Date.now();
@@ -2249,6 +2248,26 @@ async function executeWithProvider(
   } catch {
     // Worktree creation failed — fall back to project root
   }
+
+  // Copy .agents directory into worktree so sandboxed providers can access
+  // agent definitions, memory, and config files. Providers like Gemini restrict
+  // file reads to the workspace directory, so these must be local.
+  let effectivePrompt = prompt;
+  if (workDir !== projectRoot) {
+    const agentsDir = join(projectRoot, '.agents');
+    const targetAgentsDir = join(workDir, '.agents');
+    if (existsSync(agentsDir) && !existsSync(targetAgentsDir)) {
+      try {
+        cpSync(agentsDir, targetAgentsDir, { recursive: true });
+      } catch {
+        // Non-fatal: agent def may still be accessible if tracked in git
+      }
+    }
+    // Rewrite absolute paths in prompt so sandboxed providers can resolve them
+    effectivePrompt = prompt.replaceAll(projectRoot, workDir);
+  }
+
+  const args = cliConfig.buildArgs(effectivePrompt);
 
   if (options.verbose) {
     writeLine(`  ${colors.dim}Provider: ${cliConfig.displayName}${RESET}`);
@@ -2291,7 +2310,7 @@ async function executeWithProvider(
     mkdirSync(logDir, { recursive: true });
   }
 
-  const escapedPrompt = prompt.replace(/'/g, "'\\''");
+  const escapedPrompt = effectivePrompt.replace(/'/g, "'\\''");
   const providerArgs = cliConfig.buildArgs(escapedPrompt).map(a => `'${a}'`).join(' ');
   const shellScript = `cd '${workDir}' && ${cliConfig.command} ${providerArgs} > '${logFile}' 2>&1`;
   const wrapperScript = `echo $$ > '${pidFile}'; ${shellScript}`;

--- a/src/lib/llm-clis.ts
+++ b/src/lib/llm-clis.ts
@@ -67,7 +67,7 @@ export const LLM_CLIS: Record<string, CLIConfig> = {
     displayName: 'Google',
     command: 'gemini',
     install: 'npm i -g @google/gemini-cli',
-    buildArgs: (prompt) => ['--prompt', prompt],
+    buildArgs: (prompt) => ['--yolo', '--prompt', prompt],
   },
 
   openai: {

